### PR TITLE
Fix iOS signin code to make it consistent with Android side

### PR DESCRIPTION
**Pull request by mehmetf**
- iOS should not return exception code. Android returns "exception" as code.
- Return null success when user cancels signin

### CODE REVIEW

File: packages/google_sign_in/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -297,7 +297,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "diff \"${PODS_ROOT}/../Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n";
+			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n";
```
Comment (tvolkert) [inline packages/google_sign_in/example/ios/Runner.xcodeproj/project.pbxproj]: Is this change intended?

Comment (mehmetf) [inline packages/google_sign_in/example/ios/Runner.xcodeproj/project.pbxproj]: No idea what it is. I will revert it.
